pkg/virt-handler/vm_test.go

```diff
@@ -3301,6 +3304,245 @@ var _ = Describe("VirtualMachineInstance", func() {
+			Expect(vmi.Status.MigrationState.Completed).To(Equal(d.Spec.Metadata.KubeVirt.Migration.Completed))
+			Expect(vmi.Status.MigrationState.Failed).To(Equal(d.Spec.Metadata.KubeVirt.Migration.Failed))
+			Expect(vmi.Status.MigrationState.AbortStatus).To(Equal(v1.MigrationAbortStatus(
+				d.Spec.Metadata.KubeVirt.Migration.AbortStatus)))
+			Expect(vmi.Status.MigrationState.FailureReason).To(Equal(d.Spec.Metadata.KubeVirt.Migration.FailureReason))
+			testutils.ExpectEvent(recorder, v1.Migrated.String())
+		})
+	})
+
+	Context("handleMigrationAbort", func() {
+		DescribeTable("should abort the migration with an abort request", func(vmi *v1.VirtualMachineInstance) {
+			client.EXPECT().CancelVirtualMachineMigration(vmi)
+			Expect(controller.handleMigrationAbort(vmi, client)).To(Succeed())
+			testutils.ExpectEvent(recorder, VMIAbortingMigration)
+		},
+			Entry("when the request failed", libvmi.New(libvmi.WithUID(vmiTestUUID),
+				libvmistatus.WithStatus(libvmistatus.New(
+					libvmistatus.WithMigrationState(v1.VirtualMachineInstanceMigrationState{
+						AbortRequested: true,
+						AbortStatus:    v1.MigrationAbortFailed,
+					})))),
+			),
+			Entry("when the request the abort status isn't set", libvmi.New(libvmi.WithUID(vmiTestUUID),
+				libvmistatus.WithStatus(libvmistatus.New(
+					libvmistatus.WithMigrationState(v1.VirtualMachineInstanceMigrationState{
+						AbortRequested: true,
+					})))),
+			),
+		)
+		DescribeTable("should do nothing", func(vmi *v1.VirtualMachineInstance) {
+			Expect(controller.handleMigrationAbort(vmi, client)).To(Succeed())
+		},
+			Entry("when the request succeeded", libvmi.New(libvmi.WithUID(vmiTestUUID),
+				libvmistatus.WithStatus(libvmistatus.New(
+					libvmistatus.WithMigrationState(v1.VirtualMachineInstanceMigrationState{
+						AbortRequested: true,
+						AbortStatus:    v1.MigrationAbortInProgress,
+					})))),
+			),
+			Entry("when the request is in progress", libvmi.New(libvmi.WithUID(vmiTestUUID),
+				libvmistatus.WithStatus(libvmistatus.New(
+					libvmistatus.WithMigrationState(v1.VirtualMachineInstanceMigrationState{
+						AbortRequested: true,
+						AbortStatus:    v1.MigrationAbortSucceeded,
+					})))),
+			),
+		)
+
+		It("should return an error if the migration cancellation failed", func() {
+			vmi := libvmi.New(libvmi.WithUID(vmiTestUUID),
+				libvmistatus.WithStatus(libvmistatus.New(
+					libvmistatus.WithMigrationState(v1.VirtualMachineInstanceMigrationState{
+						AbortRequested: true,
+					}))))
+			client.EXPECT().CancelVirtualMachineMigration(vmi).Return(fmt.Errorf("some error"))
+			Expect(controller.handleMigrationAbort(vmi, client)).To(HaveOccurred())
+		})
+	})
+
+	Context("claimDeviceOwnership", func() {
+		var path string
+		BeforeEach(func() {
+			path = GinkgoT().TempDir()
+		})
+
+		Context("with a generic device", func() {
+			const device = "device"
+			BeforeEach(func() {
+				Expect(os.Mkdir(filepath.Join(path, "dev"), 0755)).To(Succeed())
```

Comment:
```suggestion
				Expect(os.Mkdir(filepath.Join(path, "dev"), 0o755)).To(Succeed())
```

nit: Linter would like to see it like this